### DISCUSSION TITLE
How to access services' properties in a plugin?

### DISCUSSION THREAD

**Discussion by fenix-hub** [Kong/kong#9438] [Help] [answered]
I'm writing a custom plugin where I'd like to access some properties of the target service, for example the `tags` of the service a client is calling to enable/disable custom logic.
The plugin is enabled globally, so every service uses it, but I'd like to customize some logics of the plugin based on a service configuration, using the `tags` property since it's very useful.
I've looked for `kong.service.tags` but can't find anything. The documetation is not that helpful for plugin development on this aspect.

**Comment by ms2008** [comment, marked as answer]
The *service/route* information for the current request is stored in the `ngx.ctx`, which you can access by doing this: `ngx.ctx.service.tags` or `ngx.ctx.route.tags`

**Reply by fenix-hub** [reply]
Thank you very much. Where to find this in the docs?

**Reply by ms2008** [reply]
https://docs.konghq.com/gateway/latest/plugin-development/pdk/kong.router/, as I think that particular property isn’t too well documented and is hard to find, we may even want to consider adding more clarifications to the documentation.

**Reply by fenix-hub** [reply]
Do you have any reference on the `balancer_address` perhaps? Since 3.x it's not accessible anymore through `ngx.ctx.balancer_address`

**Reply by ms2008** [reply]
Ref: https://github.com/Kong/kong/pull/9043 😊
